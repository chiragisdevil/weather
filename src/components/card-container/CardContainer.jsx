import React from "react";
import WeatherCard from "../weather-card/WeatherCard";
import "./card-container.scss";
import moment from "moment";

function CardContainer({forecast}){
    console.log(forecast); 
    let dailyRecord = null;
    if (forecast){
        dailyRecord = forecast.filter((item, index) => (index % 8 === 0))
    } 
    
    return(
        <div className="card-list">
            {dailyRecord? 
                dailyRecord.map(
                    (cardItem) => <WeatherCard  
                                key={moment.unix(cardItem.dt).format('dddd')}
                                day={moment.unix(cardItem.dt).format('dddd')}
                                Min={cardItem.main.temp_min}
                                Max={cardItem.main.temp_max}
                                icon={cardItem.weather[0].icon}
                                description = {cardItem.weather[0].description} 
                />):
                null
            }
            
        </div>
    )
}

export default CardContainer;

