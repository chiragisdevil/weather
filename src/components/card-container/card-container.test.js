// import { EnzymeAdapter } from "enzyme";
import {shallow} from "enzyme";
import CardContainer from "./CardContainer";
import sampleTestData from "./card-container-sample";

describe("Card Component tests", () => {

it("checks if card container component matches the snapshot", () => {
    expect(shallow(<CardContainer />)).toMatchSnapshot();
})


it("checks if card container component matches the snapshot", () => {
    expect(shallow(<CardContainer forecast={sampleTestData}/>)).toMatchSnapshot();
})


});
