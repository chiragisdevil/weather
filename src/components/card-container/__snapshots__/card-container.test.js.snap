// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component tests checks if card container component matches the snapshot 1`] = `ShallowWrapper {}`;

exports[`Card Component tests checks if card container component matches the snapshot 2`] = `ShallowWrapper {}`;
